### PR TITLE
Networking support

### PR DESCRIPTION
**Pull request by rajatchopra**
This PR adds networking support for ocid using CNI plugin interface. Any plugin that conforms to CNI will be picked up by ocid and appropriate hooks will be called. In the case of no plugin being present, a noop plugin is returned.
TODO: 
1. podNamespace needs to be provided to the cni calls
2. Perhaps make the env var names programmable
3. Double/Triple testing :)

@mrunalp PTAL. Thanks.

### CODE REVIEW

File: server/runtime.go [hunk 1]
```diff
@@ -148,6 +148,16 @@ func (s *Server) CreatePodSandbox(ctx context.Context, req *pb.CreatePodSandboxR
 		return nil, err
 	}
 
+	// Setup the network
+	podNamespace := ""
+	netnsPath, err := container.NetNsPath()
```
Comment (mrunalp) [inline server/runtime.go:157]: We need to update status before we ask for the network namespace path:

```
       if err := s.runtime.UpdateStatus(container); err != nil {
               return nil, err
       }
```

Comment (rajatchopra) [reply server/runtime.go:157]: Fixed

**Comment by runcom** [comment]
https://github.com/mrunalp/ocid/pull/38/commits/e6317b58d29e47e4a9b2de5f800f87bbf0a9c110 looks good to me

**Comment by mrunalp** [comment]
@rajatchopra The commits need to be signed.

**Comment by rajatchopra** [comment]
e6317b5 is now 3f53ae9. Signed off appropriately.
attention: @mrunalp

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**